packages/pillarbox-playlist/src/pillarbox-playlist.js

```diff
@@ -0,0 +1,292 @@
+import pillarbox from '@srgssr/pillarbox-web';
+
+/**
+ * @ignore
+ * @type {typeof import('video.js/dist/types/plugin').default}
+ */
+const Plugin = pillarbox.getPlugin('plugin');
+const log = pillarbox.log.createLogger('pillarbox-playlist');
+
+/**
+ * Represents a Plugin that allows control over a playlist.
+ */
+class PillarboxPlaylist extends Plugin {
+  /**
+   * The items in the playlist.
+   *
+   * @type {PlaylistItem[]}
+   * @private
+   */
+  items_ = [];
+  /**
+   * The current index.
+   *
+   * @type {number}
+   * @private
+   */
+  currentIndex_ = -1;
+  /**
+   * Whether the repeat is enabled or not. If repeat is enabled once the last
+   * element of the playlist ends the next element will be the first one. This
+   * mode only works forwards, i.e. when advancing to the next element.
+   *
+   * @type boolean
+   */
+  repeat = false;
+  /**
+   * Whether auto-advance is enabled or not.
+   *
+   * @type boolean
+   */
+  autoadvance = false;
+
+  /**
+   * Handles the 'ended' event when triggered. This method serves as a proxy to
+   * the main `ended` handler, ensuring that additional logic can be executed or
+   * making it easier to detach the event listener later.
+   *
+   * @private
+   */
+  onEnded_ = () => this.handleEnded();
+
+  /**
+   * Creates an instance of a pillarbox playlist.
+   *
+   * @param {import('@srgssr/pillarbox-web').Player} player The player instance.
+   * @param {Object} options Configuration options for the plugin.
+   */
+  constructor(player, options) {
+    super(player);
+    if (options.playlist && options.playlist.length) {
+      player.ready(() => {
+        this.load(...options.playlist);
+      });
+    }
+    this.autoadvance = !!options.autoadvance;
+    this.repeat = !!options.repeat;
+    this.player.on('ended', this.onEnded_);
+  }
+
+  dispose() {
+    this.player.off('ended', this.onEnded_);
+  }
+
+  /**
+   * Loads a playlist into the player. This method will load the first element
+   * in the playlist. Use it to initialize the playlist.
+   *
+   * @param {...PlaylistItem} items The playlist items to load.
+   */
+  load(...items) {
+    this.items_ = items;
+    this.select(0);
+  }
```

Comment:
Why not just using a good old array instead of the spread operator ? In terms of Dx that means a developer will have to do something like:

```javascript
playlistplugin.load({}, {}, {})
```
or

```javascript
const playlistItems = [{},{},{}];

playlistplugin.load(...playlistItems);
```

Reply:
The idea is to avoid external changes in the array affecting the internal playlist. If we don't use the spread operator in the parameter, we would need to do it internally, like this:

```javascript
load(items) {
    this.items_ = [...items];
    this.select(0);
}
```

While it’s less explicit in the declaration, it achieves the same goal. We can mention in the documentation that a copy of the array is made internally to ensure immutability.